internal/namespaces/container/v1beta1/custom_deploy.go

```diff
@@ -0,0 +1,355 @@
+package container
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	dockertypes "github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/fatih/color"
+	"github.com/scaleway/scaleway-cli/v2/internal/core"
+	"github.com/scaleway/scaleway-cli/v2/internal/namespaces/container/v1beta1/containerutils"
+	"github.com/scaleway/scaleway-cli/v2/internal/tasks"
+	"github.com/scaleway/scaleway-cli/v2/internal/terminal"
+	container "github.com/scaleway/scaleway-sdk-go/api/container/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type containerDeployRequest struct {
+	Region scw.Region
+
+	Name        string
+	Dockerfile  string
+	BuildSource string
+	Cache       bool
+
+	NamespaceID *string
+	Port        uint32
+}
+
+func containerDeployCommand() *core.Command {
+	return &core.Command{
+		Short:     `Deploy a container`,
+		Long:      `Automatically build and deploy a container.`,
+		Namespace: "container",
+		Resource:  "deploy",
+		Groups:    []string{"workflow"},
+		ArgsType:  reflect.TypeOf(containerDeployRequest{}),
+		ArgSpecs: core.ArgSpecs{
+			{
+				Name:  "name",
+				Short: "Name of the application",
+				Default: func(ctx context.Context) (value string, doc string) {
+					currentDirection, err := os.Getwd()
+					if err != nil {
+						return "", ""
+					}
+
+					name := filepath.Base(currentDirection)
+					if name == "." {
+						return "", ""
+					}
+
+					name = "app-" + name
+					return name, name
+				},
+			},
+			{
+				Name:    "dockerfile",
+				Short:   "Path to the Dockerfile",
+				Default: core.DefaultValueSetter("Dockerfile"),
+			},
+			{
+				Name:    "build-source",
+				Short:   "Path to the build context",
+				Default: core.DefaultValueSetter("."),
+			},
+			{
+				Name:    "cache",
+				Short:   "Use cache when building the image",
+				Default: core.DefaultValueSetter("true"),
+			},
+			{
+				Name:    "port",
+				Short:   "Port to expose",
+				Default: core.DefaultValueSetter("8080"),
+			},
+			{
+				Name:  "namespace-id",
+				Short: "Container Namespace ID to deploy to",
+			},
+			core.RegionArgSpec(scw.RegionFrPar, scw.RegionNlAms, scw.RegionPlWaw, scw.Region(core.AllLocalities)),
+		},
+		Run: containerDeployRun,
+	}
+}
+
+func containerDeployRun(ctx context.Context, argsI interface{}) (i interface{}, e error) {
+	args := argsI.(*containerDeployRequest)
+
+	client := core.ExtractClient(ctx)
+	api := container.NewAPI(client)
+
+	fileInfo, err := os.Stat(args.Dockerfile)
+	if err != nil {
+		return nil, fmt.Errorf("could not stat '%s': %v", args.Dockerfile, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("could not open '%s': %v", args.Dockerfile, err)
```

stat might not be clear for everybody